### PR TITLE
#62 Encapsulation Commit

### PR DESCRIPTION
**Pull request by ivelin980**

### CODE REVIEW

File: javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/shopping/spree/ShoppingSpree.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+			scan.close();
+			return;
+		}
+		String[] shopping = scan.nextLine().split("\\s+");
+		while (!"END".equalsIgnoreCase(shopping[0])) {
+			getPersonByName(people, shopping[0]).buyProduct(getProductByName(products, shopping[1]));
+			shopping = scan.nextLine().split("\\s+");
+		}
+
+		for (Person p : people) {
```
Comment (ngavrailov) [inline javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/shopping/spree/ShoppingSpree.java:58]: people.stream().forEach(System.out::println);
this way looks clearer IMO.

File: javaOOP_encapsulation/src/jv/oop/encapsulation/lab/first/and/reserve/team/Team.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	}
+	
+	public String getName() {
+		return this.name;
+	}
+	
+	public void addPlayer(Worker person) {
+		if(person.getAge()<40){
+			firstTeam.add(person);
+		}
```
Comment (ngavrailov) [inline javaOOP_encapsulation/src/jv/oop/encapsulation/lab/first/and/reserve/team/Team.java:27]: formatting

File: javaOOP_encapsulation/src/jv/oop/encapsulation/lab/salary/increase/Worker.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import jv.oop.encapsulation.lab.sort.by.name.and.age.Person;
+
+public class Worker extends Person {
+
+	private double salary;
+
+	public Worker(String firstName, String lastName, int age, double salary) {
+		super(firstName, lastName, age);
+		setSalary(salary);
+		}
```
Comment (ngavrailov) [inline javaOOP_encapsulation/src/jv/oop/encapsulation/lab/salary/increase/Worker.java:12]: formatting

File: javaOOP_encapsulation/src/jv/oop/encapsulation/lab/sort/by/name/and/age/SortByNameAndAge.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		Collections.sort(people, (firstPerson, secondPerson) -> {
+			int sComp = firstPerson.getFirstName().compareTo(secondPerson.getFirstName());
+			if (sComp != 0) {
+				return sComp;
+			} else {
+				return Integer.compare(firstPerson.getAge(), secondPerson.getAge());
+			}
+		});
+
+		for (Person person : people) {
```
Comment (ngavrailov) [inline javaOOP_encapsulation/src/jv/oop/encapsulation/lab/sort/by/name/and/age/SortByNameAndAge.java:46]: Print with stream. Looks better for me.

File: javaOOP_encapsulation/src/jv/oop/encapsulation/lab/validation/data/ValidationData.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+		for (int i = 0; i < n; i++) {
+			String[] input = reader.readLine().split(" ");
+			try {
+				people.add(new Worker(input[0], input[1], Integer.parseInt(input[2]), Double.parseDouble(input[3])));
+			} catch (Exception e) {
+				System.out.println(e.getMessage());
+				return;
+			}
+		}
+		for (Worker person : people) {
```
Comment (ngavrailov) [inline javaOOP_encapsulation/src/jv/oop/encapsulation/lab/validation/data/ValidationData.java:37]: Print with stream. Looks better for me.

**Review by ngavrailov — CHANGES_REQUESTED** [review]

File: javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/pizza/calories/Topping.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
+package jv.oop.encapsulation.exercise.pizza.calories;
+
+import java.util.Map;
+
+public class Topping {
+	private static final Map<String,Double> TOPPINGS_WITH_MODIFIERS = Map.of("Meat",1.2,"Veggies",0.8,"Cheese",1.1,"Sauce",0.9);
```
[the remark this answers is not in the document]
Comment (ivelin980) [reply javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/pizza/calories/Topping.java:6]: it is not necessary

File: javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/pizza/calories/Dough.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
+package jv.oop.encapsulation.exercise.pizza.calories;
+
+import java.util.Map;
+
+public class Dough {
+	private static final Map<String,Double> FLOUR_TYPES_WITH_MODIFIERS = Map.of("White",1.5,"Wholegrain",1.0);
```
[the remark this answers is not in the document]
Comment (ivelin980) [reply javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/pizza/calories/Dough.java:6]: It is not necessary

File: javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/pizza/calories/Topping.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+	
+	private void setWeight(double weight) {
+		if(weight < 1 || weight > 50) {
+			throw new IllegalArgumentException(this.toppingType + " weight should be in the range[1..50].");
+		}
+		this.weight=weight;
+	}
+	
+	public double calculateCalories() {
+		return weight*2*TOPPINGS_WITH_MODIFIERS.get(toppingType);
```
[the remark this answers is not in the document]
Comment (ivelin980) [reply javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/pizza/calories/Topping.java:31]: fixed

[the remark this answers is not in the document]
Comment (ivelin980) [reply javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/pizza/calories/Topping.java:31]: fixed

File: javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/shopping/spree/Person.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+		}
+	}
+
+	public String getName() {
+		return this.name;
+	}
+	
+	@Override
+	public String toString() {
+		String productsOutput = products.isEmpty() ? "Nothing bought" : products.stream().map(p -> p.getName()).collect(Collectors.joining(", "));
```
[the remark this answers is not in the document]
Comment (ivelin980) [reply javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/shopping/spree/Person.java:57]: fixed

File: javaOOP_encapsulation/src/jv/oop/encapsulation/lab/first/and/reserve/team/Team.java (hunk 2 above)
Comment (ivelin980) [reply javaOOP_encapsulation/src/jv/oop/encapsulation/lab/first/and/reserve/team/Team.java:27]: fixed

File: javaOOP_encapsulation/src/jv/oop/encapsulation/lab/salary/increase/Worker.java (hunk 3 above)
Comment (ivelin980) [reply javaOOP_encapsulation/src/jv/oop/encapsulation/lab/salary/increase/Worker.java:12]: fixed

File: javaOOP_encapsulation/src/jv/oop/encapsulation/lab/sort/by/name/and/age/SortByNameAndAge.java (hunk 4 above)
Comment (ivelin980) [reply javaOOP_encapsulation/src/jv/oop/encapsulation/lab/sort/by/name/and/age/SortByNameAndAge.java:46]: fixed

File: javaOOP_encapsulation/src/jv/oop/encapsulation/lab/validation/data/ValidationData.java (hunk 5 above)
Comment (ivelin980) [reply javaOOP_encapsulation/src/jv/oop/encapsulation/lab/validation/data/ValidationData.java:37]: fixed

File: javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/animal/farm/AnimalFarm.java [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+	public static void main(String[] args) {
+		Scanner scan = new Scanner(System.in);
+		String name = scan.nextLine();
+		int age = Integer.parseInt(scan.nextLine());
+		try {
+			Chicken chicken = new Chicken(name, age);
+			System.out.println(chicken);
+		} catch (Exception e) {
+			System.out.println(e.getMessage());
+			scan.close();
```
[the remark this answers is not in the document]
Comment (ivelin980) [reply javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/animal/farm/AnimalFarm.java:31]: fixed

File: javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/classbox/Box.java [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+	
+	public void validateData(double parameter, String prefix) {
+		if(parameter <= 0) {
+			throw new IllegalArgumentException(prefix + " cannot be zero or negative");
+		}
+	}
+	
+	@Override
+	public String toString() {
+		return String.format("Surface Area - %.2f%nLateral Surface Area - %.2f%nVolume - %.2f", calculateSurfaceArea(),calculateLateralSurfaceArea(),calculateVolume());
```
[the remark this answers is not in the document]
Comment (ivelin980) [reply javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/classbox/Box.java:49]: fixed

File: javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/shopping/spree/ShoppingSpree.java (hunk 1 above)
Comment (ivelin980) [reply javaOOP_encapsulation/src/jv/oop/encapsulation/exercise/shopping/spree/ShoppingSpree.java]: fixed

**Review by ngavrailov — APPROVED** [review]

**Pull request merged**